tests/test_output_formatter.cpp

```diff
@@ -276,1056 +295,697 @@ class LocaleSettingTest : public testing::Test
+     "            \"version\": \"20190520\"\n"
+     "        },\n"
+     "        \"snapcraft:core18\": {\n"
+     "            \"aliases\": [\n"
+     "            ],\n"
+     "            \"os\": \"\",\n"
+     "            \"release\": \"Snapcraft builder for core18\",\n"
+     "            \"remote\": \"snapcraft\",\n"
+     "            \"version\": \"20190520\"\n"
+     "        }\n"
+     "    }\n"
+     "}\n",
+     "json_find_multiple_duplicate_image"},
+    {&csv_formatter, &empty_find_reply, "Image,Remote,Aliases,OS,Release,Version\n", "csv_find_empty"},
+    {&csv_formatter, &find_one_reply,
+     "Image,Remote,Aliases,OS,Release,Version\n"
+     "ubuntu,,,Ubuntu,18.04 LTS,20190516\n",
+     "csv_find_one"},
+    {&csv_formatter, &find_multiple_reply,
+     "Image,Remote,Aliases,OS,Release,Version\n"
+     "lts,,,Ubuntu,18.04 LTS,20190516\n"
+     "daily:19.10,daily,eoan;devel,Ubuntu,19.10,20190516\n",
+     "csv_find_multiple"},
+    {&csv_formatter, &find_multiple_reply_duplicate_image,
+     "Image,Remote,Aliases,OS,Release,Version\n"
+     "core18,,,Ubuntu,Core 18,20190520\n"
+     "snapcraft:core18,snapcraft,,,Snapcraft builder for core18,20190520\n",
+     "csv_find_multiple_duplicate_image"},
+    {&yaml_formatter, &empty_find_reply,
+     "errors:\n"
+     "  []\n"
+     "images:\n"
+     "  {}\n",
+     "yaml_find_empty"},
+    {&yaml_formatter, &find_one_reply,
+     "errors:\n"
+     "  []\n"
+     "images:\n"
+     "  ubuntu:\n"
+     "    aliases:\n"
+     "      []\n"
+     "    os: Ubuntu\n"
+     "    release: 18.04 LTS\n"
+     "    version: 20190516\n"
+     "    remote: \"\"\n",
+     "yaml_find_one"},
+    {&yaml_formatter, &find_multiple_reply,
+     "errors:\n"
+     "  []\n"
+     "images:\n"
+     "  lts:\n"
+     "    aliases:\n"
+     "      []\n"
+     "    os: Ubuntu\n"
+     "    release: 18.04 LTS\n"
+     "    version: 20190516\n"
+     "    remote: \"\"\n"
+     "  daily:19.10:\n"
+     "    aliases:\n"
+     "      - eoan\n"
+     "      - devel\n"
+     "    os: Ubuntu\n"
+     "    release: 19.10\n"
+     "    version: 20190516\n"
+     "    remote: daily\n",
+     "yaml_find_multiple"},
+    {&yaml_formatter, &find_multiple_reply_duplicate_image,
+     "errors:\n"
+     "  []\n"
+     "images:\n"
+     "  core18:\n"
+     "    aliases:\n"
+     "      []\n"
+     "    os: Ubuntu\n"
+     "    release: Core 18\n"
+     "    version: 20190520\n"
+     "    remote: \"\"\n"
+     "  snapcraft:core18:\n"
+     "    aliases:\n"
+     "      []\n"
+     "    os: \"\"\n"
+     "    release: Snapcraft builder for core18\n"
+     "    version: 20190520\n"
+     "    remote: snapcraft\n",
+     "yaml_find_multiple_duplicate_image"}};
+} // namespace
+
+TEST_P(FormatterSuite, properly_formats_output)
+{
+    const auto param = GetParam();
+    const auto formatter = std::get<0>(param);
+    const auto reply = std::get<1>(param);
+    const auto expected_output = std::get<2>(param);
```

Comment:
Did you mean to copy `param` and then its contents? Not that it matters much, I suppose...

Reply:
Moved to structured bindings now. Could be even nicer with `[[maybe_unused]]` and nesting, but not possible yet.